### PR TITLE
[ALLUXIO-2691] Separate BlockMaster interface and impl

### PR DESCRIPTION
**Pull request by apc999**
https://alluxio.atlassian.net/browse/ALLUXIO-2691

### CODE REVIEW

**Comment by apc999** [comment]
Jenkins, test this please

File: core/server/master/src/test/java/alluxio/master/AlluxioMasterRestServiceHandlerTest.java [hunk 1]
```diff
@@ -116,7 +117,7 @@ public void before() throws Exception {
     mClock = new ManualClock();
     mExecutorService =
         Executors.newFixedThreadPool(2, ThreadFactoryUtils.build("TestBlockMaster-%d", true));
-    mBlockMaster = new BlockMaster(registry, factory, mClock,
+    mBlockMaster = new BlockMasterFactory().create(registry, factory, mClock,
```
Comment (jsimsa) [inline core/server/master/src/test/java/alluxio/master/AlluxioMasterRestServiceHandlerTest.java]: this test can use the simpler constructor (`mClock` and `mExecutorService` is not used for anything)

File: core/server/master/src/test/java/alluxio/master/block/BlockMasterTest.java [hunk 2]
```diff
@@ -86,7 +86,7 @@ public void before() throws Exception {
     mClock = new ManualClock();
     mExecutorService =
         Executors.newFixedThreadPool(2, ThreadFactoryUtils.build("TestBlockMaster-%d", true));
-    mMaster = new BlockMaster(registry, factory, mClock,
+    mMaster = new BlockMasterFactory().create(registry, factory, mClock,
```
Comment (jsimsa) [inline core/server/master/src/test/java/alluxio/master/block/BlockMasterTest.java]: for consistency with FileSystemMaster, use the DefaultBlockMaster constructor here instead

File: core/server/master/src/main/java/alluxio/master/block/BlockMasterFactory.java [hunk 3]
```diff
@@ -46,9 +49,28 @@ public String getName() {
...
+  public BlockMaster create(MasterRegistry registry, JournalFactory journalFactory) {
     Preconditions.checkArgument(journalFactory != null, "journal factory may not be null");
     LOG.info("Creating {} ", BlockMaster.class.getName());
-    return new BlockMaster(registry, journalFactory);
+    return new DefaultBlockMaster(registry, journalFactory, new SystemClock(),
+        ExecutorServiceFactories
+            .fixedThreadPoolExecutorServiceFactory(Constants.BLOCK_MASTER_NAME, 2));
+  }
+
+  /**
```
Comment (jsimsa) [inline core/server/master/src/main/java/alluxio/master/block/BlockMasterFactory.java]: after the changes propose in my other comments, this is no longer needed

**Comment by apc999** [comment]
@jsimsa  PTAL

**Comment by apc999** [comment]
Jenkins, test this please

**Comment by apc999** [comment]
Jenkins, test this please

**Pull request merged**